pkg/jobs/hypershift/hypershift.go

```diff
@@ -0,0 +1,533 @@
+			klog.V(0).Info("The " + jobType + "ing job from Hypershift failed ✓")
+			/*
+				  Detect that we've failed but there's no Hypershift equivalent of ProvisionStoppedCondition.
+					The problem with trying to detect failure for hypershift is that there's no total failure state
+					where the operator will give up trying. Users can always fix an issue ie. WebIdentity error to
+					allow the provision to continue.
+			*/
+		}
+	}
+
+	if hostedCluster != nil {
+		klog.Warning(hostedCluster.Object["status"].(map[string]interface{})["conditions"])
+	}
+	return errors.New("Timed out waiting for job")
+}
+
+func isHostedReady(hostedCluster *unstructured.Unstructured, isUpgrade bool) bool {
+	status := hostedCluster.Object["status"].(map[string]interface{})
+	conditions := status["conditions"].([]interface{})
+	degraded := true
+	clusterVersionProgressing := true
+	clusterVersionProgressingMsg := ""
+	clusterVersionAvailable := false
+	clusterVersionAvailableMsg := ""
+	available := false
+	progressing := false
+
+	for _, condition := range conditions {
+		conditionType := condition.(map[string]interface{})["type"].(string)
+		conditionStatus := condition.(map[string]interface{})["status"].(string)
+
+		klog.V(4).Info("Type " + condition.(map[string]interface{})["type"].(string) +
+			" Status " + condition.(map[string]interface{})["status"].(string))
+		switch conditionType {
+		case "Degraded":
+			degraded, _ = strconv.ParseBool(conditionStatus)
+		case "ClusterVersionAvailable":
+			clusterVersionAvailable, _ = strconv.ParseBool(conditionStatus)
+			clusterVersionAvailableMsg = condition.(map[string]interface{})["message"].(string)
+		case "ClusterVersionProgressing":
+			clusterVersionProgressing, _ = strconv.ParseBool(conditionStatus)
+			clusterVersionProgressingMsg = condition.(map[string]interface{})["message"].(string)
+		case "Available":
+			available, _ = strconv.ParseBool(conditionStatus)
+		case "Progressing":
+			progressing, _ = strconv.ParseBool(conditionStatus)
+		}
+	}
+
+	klog.V(4).Info("Available " + strconv.FormatBool(available))
+	if !available {
+		return false
+	}
+	klog.V(4).Info("degraded " + strconv.FormatBool(degraded))
+	if degraded {
+		return false
+	}
+	klog.V(4).Info("clusterVersionProgressing " + strconv.FormatBool(clusterVersionProgressing))
+	if clusterVersionProgressing {
+		return false
+	}
+	klog.V(4).Info("clusterVersionAvailable " + strconv.FormatBool(clusterVersionAvailable))
+	if !clusterVersionAvailable {
+		return false
+	}
+	klog.V(4).Info("clusterVersionProgressingMsg " + clusterVersionProgressingMsg)
+	if !strings.Contains(clusterVersionProgressingMsg, "Cluster version is") {
+		return false
+	}
+	klog.V(4).Info("clusterVersionAvailableMsg " + clusterVersionAvailableMsg)
+	if !strings.Contains(clusterVersionAvailableMsg, "Done applying") {
+		return false
+	}
+	klog.V(4).Info("progressing " + strconv.FormatBool(progressing))
+	if isUpgrade && progressing {
+		return false
+	}
+
+	return true
```

Comment:
Thanks @mikeshng I think what you suggested will be better code quality so I'll make the changes. I was planning to retest anyway after all the changes are made.

Reply:
@fxiang1 let's not do the inverse logic thing ie changing the return from ready to not ready. Let's keep the general logic the same and if you want to clean it up and little maybe try what I suggested in https://github.com/stolostron/cluster-curator-controller/pull/183#discussion_r1337470256 and see if it's a bit better. I think we can still use a comment to really know what is ready.

Reply:
Sorry for the spam @fxiang1 but you will have to at least fix this function a bit.
The strconv.ParseBool(conditionStatus) is not going to work all the time because 
`ConditionUnknown = v1.ConditionUnknown`
see https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1beta1/types.go#L79
so its not just true and false

Reply:
Yep I made changes as suggested in https://github.com/stolostron/cluster-curator-controller/pull/183#discussion_r1337470256 .
Also fixed the `parseConditionStatus()` func to handle `ConditionUnknown = v1.ConditionUnknown`.